src/openrct2/ride/coaster/HybridCoaster.cpp

```diff
@@ -0,0 +1,11617 @@
+#pragma region Copyright(c) 2014 - 2017 OpenRCT2 Developers
+/*****************************************************************************
+ * OpenRCT2, an open source clone of Roller Coaster Tycoon 2.
+ *
+ * OpenRCT2 is the work of many authors, a full list can be found in contributors.md
+ * For more information, visit https//:ggithub.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * A full copy of the GNU General Public License can be found in licence.txt
+ *****************************************************************************/
+#pragma endregion
+
+#include "../../drawing/Drawing.h"
+#include "../../interface/Viewport.h"
+#include "../../paint/Paint.h"
+#include "../../paint/Supports.h"
+#include "../../sprites.h"
+#include "../../world/Map.h"
+#include "../../world/Sprite.h"
+#include "../RideData.h"
+#include "../TrackData.h"
+#include "../TrackPaint.h"
+
+static uint32_t hybrid_rc_get_track_colour(paint_session* session)
+{
+    if (session->TrackColours[SCHEME_TRACK] == 0x21600000)
+        return 0x21600000; // TODO dirty hack
+    else
+        return (session->TrackColours[SCHEME_TRACK] & ~0x1F000000) | ((session->TrackColours[SCHEME_SUPPORTS] & 0xF80000) << 5);
+}
+
+static void hybrid_rc_track_flat(
+    paint_session* session, uint16_t rideIndex, uint8_t trackSequence, uint8_t direction, int32_t height,
+    const TileElement* tileElement)
+{
+    if (tileElement->AsTrack()->HasChain())
+    {
+        switch (direction)
+        {
+            case 0:
+                sub_98197C_rotated(
```

Comment:
`sub_98197C_rotated` was designed to reduce this duplication on rotation.
Just do:
```
sub_98197C_rotated( session, direction, hybrid_rc_get_track_colour(session) | (SPR_G2_HYBRID_LIFT_TRACK_FLAT + direction), 0, 0, 32, 20, 3, height, 0, 6, height);
```
drop all of the switch.